c10/util/generic_math.h

```diff
@@ -4,6 +4,17 @@
 #include <c10/util/TypeSafeSignMath.h>
 #include <cmath>
 
+#if defined(__CUDA_ARCH__)
+#include <c10/cuda/CUDAMathCompat.h>
+#define compat_copysign c10::cuda::compat::copysign
+#elif defined(__HIPCC__)
+#include <c10/hip/HIPMathCompat.h>
+#define compat_copysign c10::hip::compat::copysign
+#else
+#include <c10/util/copysign.h>
```

Comment:
This worries me. It includes quite a few headers as a closure, which is a threat to the ABI-compatibility mode. You either need to audit all those headers, or you can separate compat_copysign-related code and move it here.

Reply:
Isn't this already covered by the fact that we don't link against libc10 in abi-compat mode?

Reply:
> Likey true for OSS, but is internal libomnibus going to hide the problem? cc @chenyang78

What ABI-compatibility tries to solve is to make any ABI-related change transparent to the generated DSO. For example, we don't want to expose or passing around c10 tensor structs across the cshim, because any changes to c10 tensor might cause mal-behaviors due to mismatches between DSO and libc10. It doesn't really matter if we link against libc10 or not. 

On the other hand, with abi-compatible cshim, we force all the changes are within libc10 so that we would haven't the aforementioned mismatches. This is why we enforce self-contained header-only files to be included in the generated wrapper code, which won't cause such inconsistency across the cshim boundary.

Reply:
I am not sure how libomnibus plays into this. It sounds like it's something that we create in the course of `buck build`; but we aren't using buck to directly compile the `model.so`s, right? I think that's all done through our Python code?

Reply:
> I remember during the early AOTI bringup @chenyang78 had to link against libomnibus instead of libtorch. Not sure about the latest status, if @chenyang78 can confirm this is not the case anymore, I am ok with this PR.

Yeah, I don't think libomnibus is an issue for this case.